Crashlytics/UnitTests/FIRCLSExistingReportManagerTests.m

```diff
@@ -0,0 +1,214 @@
+  NSString *metadata = [NSString stringWithFormat:METADATA_FORMAT, time, reportID];
+  if (![self createMetadata:metadata forReport:report]) {
+    return nil;
+  }
+
+  if (withEvents) {
+    [self createCrashExceptionFile:@"Content doesn't matter" forReport:report];
+  }
+
+  return report;
+}
+
+- (BOOL)createFileWithContents:(NSString *)contents atPath:(NSString *)path {
+  return [self.fileManager.underlyingFileManager
+      createFileAtPath:path
+              contents:[contents dataUsingEncoding:NSUTF8StringEncoding]
+            attributes:nil];
+}
+
+- (BOOL)createMetadata:(NSString *)value forReport:(FIRCLSInternalReport *)report {
+  return [self createFileWithContents:value atPath:[report metadataPath]];
+}
+
+- (BOOL)createCrashExceptionFile:(NSString *)value forReport:(FIRCLSInternalReport *)report {
+  return [self createFileWithContents:value
+                               atPath:[report pathForContentFile:FIRCLSReportExceptionFile]];
+}
+
+- (BOOL)reportPathAtIndex:(NSUInteger)index isReportID:(NSString *)reportID {
+  return [[self.existingReportManager.existingUnemptyActiveReportPaths objectAtIndex:index]
+      containsString:reportID];
+}
+
+#pragma mark - Tests
+
+- (void)testNoReports {
+  [self.existingReportManager collectExistingReports];
+
+  [self.existingReportManager.operationQueue waitUntilAllOperationsAreFinished];
+
+  // Reports without events should be deleted
+  XCTAssertEqual([[self contentsOfActivePath] count], 0);
+  XCTAssertEqual(self.existingReportManager.unsentReportsCount, 0);
+  XCTAssertEqual(self.existingReportManager.newestUnsentReport, nil);
+  XCTAssertEqual(self.existingReportManager.existingUnemptyActiveReportPaths.count, 0);
+}
+
+- (void)testReportNoEvents {
+  [self createActiveReportWithID:@"report_A" time:12312 withEvents:NO];
+  [self createActiveReportWithID:@"report_B" time:12315 withEvents:NO];
+
+  [self.existingReportManager collectExistingReports];
+
+  [self.existingReportManager.operationQueue waitUntilAllOperationsAreFinished];
+
+  // Reports without events should be deleted
+  XCTAssertEqual([[self contentsOfActivePath] count], 0);
+  XCTAssertEqual(self.existingReportManager.unsentReportsCount, 0);
+  XCTAssertEqual(self.existingReportManager.newestUnsentReport, nil);
+  XCTAssertEqual(self.existingReportManager.existingUnemptyActiveReportPaths.count, 0);
+}
+
+- (void)testUnsentReportsUnderLimit {
+  [self createActiveReportWithID:@"report_A" time:12312 withEvents:YES];
+  [self createActiveReportWithID:@"report_B" time:12315 withEvents:YES];
+  [self createActiveReportWithID:@"report_C" time:31533 withEvents:YES];
+  [self createActiveReportWithID:@"report_D" time:63263 withEvents:YES];
+
+  [self.existingReportManager collectExistingReports];
+
+  [self.existingReportManager.operationQueue waitUntilAllOperationsAreFinished];
+
+  // Reports with events should be kept if there's less than MAX_UNSENT_REPORTS reports
+  XCTAssertEqual([[self contentsOfActivePath] count], 4);
```

Comment:
If `MAX_UNSENT_REPORTS` is extern'ed can we reference it here and below to clarify why the expected value is 4?